### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by Fahim-Khan-P**
- [ ] Install CanCanCan in your project.
- [ ] Add a role column to the users table. Remember to use a migration for this.
- [ ] A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
- [ ] For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- [ ] A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
- [ ] For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -32,6 +32,11 @@ def create_comment
...
   end
 
+  def destroy
+    Post.delete(params[:id])
+    redirect_to user_posts_path
+  end
+
   def create_like
     @like = Like.new(author_id: current_user.id, post: Post.find(params[:id]))
     redirect_to user_post_path if @like.save
```
Comment (FrederickMih) [inline app/controllers/posts_controller.rb:42]: - [ ] Hi @Fahim-Khan-P, we are equally supposed to use CanCanCan for Post authorization in the Post Controller as below:

```ruby
 class PostsController < ApplicationController
    load_and_authorize_resource

    ...
```
- Let's kindly fix it.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand:  @Fahim-Khan-P, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- PR has a good title and summary :+1: 
- Linter checks passed :heavy_check_mark: 


## Required Changes ♻️

- [ ] Please let's kindly resolve the issue of duplicate users tables as seen in the screenshot below:
Screenshot from 2023-05-05 21-21-22


_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Fahim-Khan-P** [comment]
### **Hi dear Reviewer😉!**

- [ ] As per your previous review, I solved the duplication of the user table you can check the schema file as well, you can see it in below **screenshot**!
jjj

**Review by Ismailco — APPROVED** [review]
### Hi @Fahim-Khan-P, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions
- [ ] I can't seem to log in to your app when signing I got redirected to this error 👇🏼 , kindly consider fixing it so your app works properly

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**